File: Day1/Pascal Triangle/main.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> ans=new ArrayList<>();
        List<Integer> prev=null;
        
         for(int i=0;i<numRows;i++){
            // by default keeping all values to be 1 , as first and last vaue will be 1 only
            List<Integer> temp=new ArrayList<>();
            
            for(int j=0;j<=i;j++){
                // for first an dlast index we have to append 1
                // In java list we cant get the element via index , rather we have to do list.get(i) to get particukar index value
                if(j==0 || j==i){
                    temp.add(1);
                    continue;
                }
                 // Take last row two values sum and update 
                temp.add(prev.get(j-1)+prev.get(j));
            }
             
            prev=temp;
            ans.add(temp);
        }
        
        return ans;
    }
}

File: Day1/Pascal Triangle/main.cpp
/*
Approach :

1. Simple implementation problem
2. Keep interating till n(given input) and calculate the 
current row value on the basis of last row two values summation

Time Complexity : O(n^2)
Space Complexity :O(n^2)

*/


class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> ans;
        
        for(int i=0;i<numRows;i++){
            // by default keeping all values to be 1 , as first and last vaue will be 1 only
            vector<int> temp(i+1,1);
            
            for(int j=1;j<i;j++){
                // Take last row two values sum and update 
                temp[j]=ans[i-1][j-1]+ans[i-1][j];
            }
            
            ans.push_back(temp);
        }
        
        return ans;
    }
};
